rust/worker/src/segment/types.rs

```diff
@@ -1,24 +1,108 @@
-use std::collections::HashMap;
-use std::sync::atomic::AtomicU32;
-use std::sync::Arc;
-
-use crate::blockstore::key::KeyWrapper;
 use crate::errors::{ChromaError, ErrorCodes};
 use crate::execution::data::data_chunk::Chunk;
-use crate::index::metadata::types::MetadataIndexError;
 use crate::types::{
-    merge_update_metadata, update_metdata_to_metdata, BooleanOperator, LogRecord, Metadata,
-    MetadataType, MetadataValue, MetadataValueConversionError, Operation, OperationRecord, Where,
-    WhereClauseComparator, WhereComparison,
+    DeletedMetadata, LogRecord, Metadata, MetadataValue, MetadataValueConversionError, Operation,
+    OperationRecord, UpdateMetadata, UpdateMetadataValue,
 };
-use crate::utils::{merge_sorted_vecs_conjunction, merge_sorted_vecs_disjunction};
 use async_trait::async_trait;
-use futures::future::BoxFuture;
-use roaring::RoaringBitmap;
+use std::collections::{HashMap, HashSet};
+use std::sync::atomic::AtomicU32;
+use std::sync::Arc;
 use thiserror::Error;
 
 use super::record_segment::{ApplyMaterializedLogError, RecordSegmentReader};
 
+// Materializes metadata from update metadata, populating the delete list
+// and upsert list.
+pub(crate) fn materialize_update_metadata(
```

Comment:
I find materialize_update_metadata + merge_update_metadata in tandem very confusing.